client/utils/account-fees.tsx

```diff
@@ -85,19 +85,41 @@ export const getCurrentBaseFee = (
 		: accountFees.base;
 };
 
+// If the current fee doesn't have a fixed or percentage rate, use the base fee's rate. Eg. when there is a promotional discount fee applied.
+const getCurrentFeeWithBaseFallback = (
+	currentFee: BaseFee | DiscountFee,
+	baseFee: BaseFee
+) => {
+	if ( ! currentFee.percentage_rate ) {
```

Comment:
This logic is incorrect: it kind of blends fees for some cases into something weird.
We want to pick the base fee instead - from this angle we don't need the function at all.

Reply:
Simplified and made amends here. I hope this is better. 

6db016a